Adding max_padded_size to to_padded_tensor

mjanderson09 opened the pull request:
Summary:
- Adding max_padded_size argument to to_padded_tensor
- Modified add_padding_kernelLauncher and kernels to iterate over padded tensor batch size instead of nested tensor batch size
- No fast path for CPU version

Test Plan:
buck test mode/dev-nosan  //caffe2/test:nested

Performance test using N1763981:

{F728168808}

Differential Revision: D36056902

cpuhrsch reviewed (approved):
Approving to unblock workflow, but note that this is stacked on top of https://github.com/pytorch/pytorch/pull/76470 so please land that first.

mjanderson09 committed:
Nested Tensor Elementwise

Differential Revision: D35773182

fbshipit-source-id: 003b2b071ef14a43f9c568fb503f8fa494f29e3b

mjanderson09 committed:
Adding output_size to to_padded_tensor (#76640)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/76640

- Adding output_size argument to to_padded_tensor
- Modified add_padding_kernelLauncher and kernels to iterate over padded tensor batch size instead of nested tensor batch size
- No fast path for CPU version

Test Plan:
buck test mode/dev-nosan  //caffe2/test:nested

Performance test using N1763981:

{F728168808}

Reviewed By: cpuhrsch

Differential Revision: D36056902

fbshipit-source-id: b9d3831a3b411f4889bfef4cf8a83706386e6402
